tests/dataclass_test.py

```diff
@@ -0,0 +1,77 @@
+#  Copyright (c) 2023 Mira Geoscience Ltd.
+#
+#  This file is part of geoapps-utils package.
+#
+#  All rights reserved.
+#
+
+from __future__ import annotations
+
+import pytest
+from geoh5py.workspace import Workspace
+from pydantic import ValidationError
+
+from geoapps_utils.driver.data import BaseData
+
+
+def test_dataclass_valid_values(tmp_path):
+    workspace = Workspace(tmp_path / "test.geoh5")
+
+    valid_params = {
+        "monitoring_directory": workspace.h5file,
+        "workspace_geoh5": workspace.h5file,
+        "geoh5": workspace.h5file,
+        "run_command": "test.driver",
+        "title": "test title",
+        "conda_environment": "test_env",
+        "conda_environment_boolean": True,
+        "workspace": workspace,
+        "run_command_boolean": False,
+    }
+
+    try:
+        model = BaseData(**valid_params)
+    except ValidationError:
+        pytest.fail()
```

Comment:
**suggestion:** let pytest deal with the exception

It will provide more details in the test report in case of failure.

```suggestion
    model = BaseData(**valid_params)
```